### ISSUE TITLE
ReferenceError: JSREPL is not defined

### ISSUE THREAD

**Issue by paazmaya** [GitbookIO/plugin-exercises#6]
While running locally, with `gitbook serve`, the page in which an exercise is being used gives the error `ReferenceError: JSREPL is not defined` which points to https://github.com/GitbookIO/plugin-exercises/blob/master/book/exercises.js#L7

At the "production" site, gitbook.io, the page works, and thus assuming that the given variable is present.

Perhaps there should be a check whether it is not available, use a reasonable default?

**Comment by ajblum** [comment]
I know this is a year old, but I'm hitting this as well.  It seems that the local `gitbook build` builds html with this script being loaded at the end:
```
        <script src="../gitbook/plugins/gitbook-plugin-exercises/jsrepl/jsrepl.js" id="jsrepl-script"></script>
```

When I manually move `jsrepl-script` above the `exercises.js` script, the interactive exercise works like:
```
    <script src="../gitbook/plugins/gitbook-plugin-exercises/jsrepl/jsrepl.js" id="jsrepl-script"></script>
    <script src="../gitbook/plugins/gitbook-plugin-exercises/exercises.js"></script>
```

Any hints where in the `gitbook build` code for the exercises plugin this insertion occurs ?  Seems like we need for it to insert the `jsrepl-script` BEFORE the `exercises.js` script line.

**Comment by ajblum** [comment]
Ok. I think I found a potential solution..at least this is what I'm going with.

After doing the `gitbook install` to install the exercises plugin, I changed *body:end* to *body:start* in `node_modules/gitbook-plugin-exercises/index.js`

```
        html: {
            "body:start": function(options) {
                return '<script src="'+options.staticBase+'/plugins/gitbook-plugin-exercises/jsrepl/jsrepl.js" id="jsrepl-script"></script>';
```

The result is, the jsrepl-script is inserted as the first line in the <body> in the rendered html.

**Comment by SamyPesse** [comment]
@ajblum can you make a PR for this change? (that will make the plugin compatible with latest versions of GitBook)

**Comment by poliveira89** [comment]
This happens with the version online. How can I do that change the order on the online version?

**Comment by AaronO** [comment]
We could merge PR #11, but ideally all JS should be loaded after `</body>`, so #11 seems like a quickfix more than a fix. Thoughts @SamyPesse ?

**Comment by jpreynat** [comment]
Fixed by 5fcfaa4f138cfa101d316d42423f09028f0e9027.
This issue has been fixed with version 2.1.0, compatible with GitBook 3.
The `jsrepl.js`script is now included first using templating.

![plugin-exercises](https://cloud.githubusercontent.com/assets/7927876/14634263/7a924a76-0621-11e6-9c93-65b7df60e93b.gif)

Thanks for your feedback.
